protocols/gossipsub/src/types.rs

```diff
@@ -197,8 +241,8 @@ pub enum SubscriptionAction {
 }
 
 #[derive(Debug, Clone, PartialEq, Eq, Hash)]
-pub struct PeerInfo {
-    pub peer_id: Option<PeerId>,
+pub(crate) struct PeerInfo {
+    pub(crate) peer_id: Option<PeerId>,
```

Comment:
```suggestion
pub struct PeerInfo {
    pub peer_id: Option<PeerId>,
```
`PeerInfo` isn't re-exported from lib.rs, so I don't think this is needed?

Reply:
Nvm, is is reaquired because of the "unreachable pub item" lint.